rhel/vex/parser.go

```diff
@@ -290,8 +271,46 @@ func (c *creator) knownAffectedVulnerabilities(ctx context.Context, v csaf.Vulne
 			return nil, fmt.Errorf("could not unbind cpe: %s %w", ch, err)
 		}
 		vuln.Repo = c.rc.Get(wfn)
+		// It is possible that we will not find a pURL, in that case
+		// the package.Name will be reported as-is.
+		purlHelper, ok := compProd.IdentificationHelper["purl"]
+		if ok {
+			purl, err := packageurl.FromString(purlHelper)
+			if err != nil {
+				zlog.Warn(ctx).
+					Str("purl", purl.String()).
```

Comment:
There are times when `FromString` returns an empty `purl` when there is an error. I think the original message of `purlHelper` was better